plugins/webp-uploads/tests/test-picture-element.php

```diff
@@ -125,4 +140,121 @@ public function data_provider_it_should_maybe_wrap_images_in_picture_element():
 			),
 		);
 	}
+
+	/**
+	 * @dataProvider data_provider_test_image_sizes_equality
+	 *
+	 * @param Closure|null $add_filter The filter.
+	 */
+	public function test_img_sizes_is_equal_to_picture_source_sizes_for_picture_element( ?Closure $add_filter ): void {
+		$mime_type = 'image/webp';
+		if ( ! wp_image_editor_supports( array( 'mime_type' => $mime_type ) ) ) {
+			$this->markTestSkipped( "Mime type $mime_type is not supported." );
+		}
+
+		$image = wp_get_attachment_image(
+			self::$image_id,
+			'large',
+			false,
+			array(
+				'class' => 'wp-image-' . self::$image_id,
+				'alt'   => 'Green Leaves',
+			)
+		);
+
+		if ( $add_filter instanceof Closure ) {
+			$add_filter();
+		}
+
+		$img_markup = apply_filters( 'the_content', $image );
+
+		// Apply picture element support.
+		$this->opt_in_to_picture_element();
+
+		$picture_markup = apply_filters( 'the_content', $image );
+
+		$img_processor = new WP_HTML_Tag_Processor( $img_markup );
+		$this->assertTrue( $img_processor->next_tag( array( 'tag_name' => 'IMG' ) ), 'There should be an IMG tag.' );
+		$img_sizes = $img_processor->get_attribute( 'sizes' );
+
+		$picture_processor = new WP_HTML_Tag_Processor( $picture_markup );
+
+		$picture_processor->next_tag( array( 'tag_name' => 'IMG' ) );
+		$this->assertSame( $img_sizes, $picture_processor->get_attribute( 'sizes' ), 'The IMG and Picture IMG have same sizes attributes.' );
+
+		while ( $picture_processor->next_tag( array( 'tag_name' => 'source' ) ) ) {
+			$this->assertSame( $img_sizes, $picture_processor->get_attribute( 'sizes' ), 'The IMG and Picture source have same sizes attributes.' );
+		}
+	}
+
+	/**
+	 * Data provider for it_should_maybe_wrap_images_in_picture_element.
+	 *
+	 * @return array<string, array<string, (Closure)|null>>
+	 */
+	public function data_provider_test_image_sizes_equality(): array {
+		return array(
+			'no_filter'   => array(
+				'add_filter' => null,
+			),
+			'with_filter' => array(
+				'add_filter' => static function (): void {
+					add_filter(
+						'wp_content_img_tag',
+						static function (
+							string $content
+						): string {
+							$processor = new WP_HTML_Tag_Processor( $content );
+							if ( ! $processor->next_tag( array( 'tag_name' => 'img' ) ) ) {
+								return $content;
+							}
```

Comment:
If there is no `IMG` in the `$content`, this would be bad, right? I don't think this should return silently.

```suggestion
						'wp_content_img_tag',
						function ( string $content ): string {
							$processor = new WP_HTML_Tag_Processor( $content );
							$this->assertTrue( $processor->next_tag( array( 'tag_name' => 'img' ) ) );
```

Reply:
It generate PHPStan error: https://github.com/WordPress/performance/pull/1354/commits/61a488f10c0b9408bdc25820a2abf2f01218ebfd